hoodie-common/src/main/java/com/uber/hoodie/common/table/log/block/metadata/HoodieLogBlockMetadata.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.uber.hoodie.common.table.log.block.metadata;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Predicate;
+
+/**
+ * Metadata abstraction for a HoodieLogBlock
+ */
+public interface HoodieLogBlockMetadata {
```

Comment:
This is the LogBlock metadata abstraction.

Reply:
Wont a simple Map<String, String> be sufficient?

Reply:
We just have 1 field, `instantTime`.. So IMO a class hierarchy for this seems an overkill. WDYT?

Reply:
I've actually added 2 fields, `instantTime` (for all blocks) and `targetInstantTime` (for command blocks). Take a look at the class `HoodieTargetInstantTime` for details.
In general, I think it's good to have this hierarchy now to allow addition of new metadata and to have well defined metadata classes. 
The current class hierarchy implementation saves space on disk too, it only writes <ordinal(int), data> for each metadata tuple. A serialized version of Map<String, String> overhead of metadata of string type for key, although that could be avoided with some intelligent code but making it complicated again.
If we agree on this, I'm planning to refactor the names of the metadata payloads to reflect something `meta`..

Reply:
The tradeoff here seems between strongly typed pojo object as value, as opposed to having an opaque string. (the problem of serializing whole string keys can be easily avoided with a enum like you already have). Agree we can extend this to arbitrary avro/pojo metadata fields in the future like a forward index possibly.

>  I'm planning to refactor the names of the metadata payloads to reflect something
Not following, are you suggesting you are going to rename the classes to something like `TargetInstantTimeLogMetadata` ?

Reply:
As per our discussion offline, I think I'm fine with reverting to using a simpler non-pojo based approach. I'll update the PR soon.
Yes, I was referring to such a rename which isn't really applicable now.